node-graph/gstd/src/memo.rs

```diff
@@ -2,35 +2,32 @@ use graphene_core::{Cache, Node};
 use once_cell::sync::OnceCell;
 
 /// Caches the output of a given Node and acts as a proxy
-pub struct CacheNode<CachedNode: Node<I>, I> {
-	node: CachedNode,
-	cache: OnceCell<CachedNode::Output>,
+pub struct CacheNode<T> {
+	cache: OnceCell<T>,
 }
-impl<'n, CashedNode: Node<I> + Copy, I> Node<I> for &'n CacheNode<CashedNode, I> {
-	type Output = &'n CashedNode::Output;
-	fn eval(self, input: I) -> Self::Output {
+impl<'n, T> Node<T> for &'n CacheNode<T> {
+	type Output = &'n T;
```

Comment:
Thoughts on returning references? The document nodes we have currently mostly take owned values meaning that down-casting will fail when references are used. Will some system be responsible for inserting clone nodes?

Reply:
When using the output of one node in multiple other nodes we need to return references to avoid cloning the image a bunch of times

Reply:
The insertion of clone would either happen at the graph rewriting level or as a generic implementation for the node.
(You could impl the document node for both &T and T and let that deal with choosing the correct implementation)